mmf/datasets/processors/bert_processors.py

```diff
@@ -206,3 +212,107 @@ def __call__(self, item):
+        else:  # self._type == "full_sentences":
+            return self._full_sentences(sentence_tokens)
+
+    def _tokenize_sentences(self, texts):
+        sentences, sentence_tokens = [], []
+        for sentence in texts:
+            sentence = sentence.strip()
+            tokens = self._tokenizer.tokenize(sentence)
+            sentences.append(sentence)
+            sentence_tokens.append(tokens)
+        return sentences, sentence_tokens
+
+    def _two_sentences(self, sentence_tokens):
+        """Concatnate only two sentences starting with either 0 or random index."""
+        """With optional separator. Truncate or else pad to max_seq_length."""
+        has_two_or_more_sentences = len(sentence_tokens) > 1
+        if "rand" in self._type:
+            start_idx = (
+                random.randint(0, len(sentence_tokens) - 2)
+                if has_two_or_more_sentences
+                else 0
+            )
+        else:
+            start_idx = 0
+        end_idx = start_idx + 2 if has_two_or_more_sentences else len(sentence_tokens)
+        return self._full_sentences(sentence_tokens[start_idx:end_idx])
+
+    def _rand_sentences(self, sentence_tokens):
+        """Concatnate all consecutive sentences with optional separator."""
+        """Start with a random sentence index aiming to fill max_seq_length."""
+        """Pad till max_seq_length."""
+        sentence_separater = self._sentence_separator("sentence")
+
+        max_start = 0
+        length_to_fullfill = self._max_seq_length - 2
+        for idx, sentence_token in enumerate(sentence_tokens):
+            if sum(len(x) for x in sentence_tokens[idx:]) < length_to_fullfill:
+                max_start = idx
+                break
+            length_to_fullfill -= len(sentence_separater)
+
+        random_idx = random.randint(0, max_start - 1 if max_start > 0 else 0)
+        return self._full_sentences(sentence_tokens[random_idx:])
+
+    def _full_sentences(self, sentence_tokens):
+        """Concatnate all sentences starting with sentence index 0."""
+        """With optional separator. Truncate or else pad to max_seq_length."""
+        tokens = []
+        lm_label_ids = []
+        for idx, sentence_token in enumerate(sentence_tokens):
+            token_a, label_a = self._random_word(
+                sentence_token, probability=self._probability
+            )
+            if idx == len(sentence_tokens) - 1:
+                tokens += token_a
+                lm_label_ids += label_a
+            else:
+                tokens += token_a + self._sentence_separator("sentence")
+                lm_label_ids += label_a + self._sentence_separator("id")
+
+        tokens = tokens[: self._max_seq_length - 2]
+        tokens = [self._CLS_TOKEN] + tokens + [self._SEP_TOKEN]
+        lm_label_ids = lm_label_ids[: self._max_seq_length - 2]
+        lm_label_ids = [-1] + lm_label_ids + [-1]
+        segment_ids = [0] * len(tokens)
```

Comment:
I think there should be a way to change `segment_id` in all this processors. Maybe through config.

Reply:
Do you mind clarifying what do you mean by "change `segment_id`"? 

Right now the default is to keep them all 0. Are you saying that we should allow the user to change the value of the segment ID through config?

Reply:
We should allow users to specify different segment ids for different sentences when they are separated with a [SEP] token or separator token.